gradle/plugins/common/src/main/kotlin/junitbuild.java-nullability-conventions.gradle.kts

```diff
@@ -27,29 +30,270 @@ nullaway {
+						"StreamToString," +
+						"StringBuilderInitWithChar," +
+						"StringJoin," +
+						"StringJoin," +
+						"SubstringOfZero," +
+						"SuppressWarningsDeprecated," +
+						"TemporalAccessorGetChronoField," +
+						"TestParametersNotInitialized," +
+						"TheoryButNoTheories," +
+						"ThreadBuilderNameWithPlaceholder," +
+						"ThrowIfUncheckedKnownChecked," +
+						"ThrowNull," +
+						"TimeZoneUsage," +
+						"TreeToString," +
+						"TryFailThrowable," +
+						"TypeParameterQualifier," +
+						"UnicodeDirectionalityCharacters," +
+						"UnicodeInCode," +
+						"UnnecessaryCheckNotNull," +
+						"UnnecessaryTypeArgument," +
+						"UnsafeWildcard," +
+						"UnusedAnonymousClass," +
+						"UnusedCollectionModifiedInPlace," +
+						"VarTypeName," +
+						"WrongOneof," +
+						"XorPower," +
+						"ZoneIdOfZ,"
+			)
+		}
 		val shouldDisableErrorProne = java.toolchain.implementation.orNull == JvmImplementation.J9
 		if (name == "compileJava" && !shouldDisableErrorProne) {
 			disable(
-
-				// This check is opinionated wrt. which method names it considers unsuitable for import which includes
-				// a few of our own methods in `ReflectionUtils` etc.
-				"BadImport",
-
-				// The findings of this check are subjective because a named constant can be more readable in many cases
-				"UnnecessaryLambda",
-
-				// Resolving findings for these checks requires ErrorProne's annotations which we don't want to use
-				"AnnotateFormatMethod",
+				// error-prone (https://errorprone.info/bugpatterns)
+				"AnnotateFormatMethod", // We don't want to use ErrorProne's annotations.
+				"BadImport", // This check is opinionated wrt. which method names it considers unsuitable for import which includes a few of our own methods in `ReflectionUtils` etc.
+				"DirectReturn", // https://github.com/junit-team/junit-framework/pull/5006#discussion_r2403984446
 				"DoNotCallSuggester",
-				"InlineMeSuggester",
 				"ImmutableEnumChecker",
-
-				// Resolving findings for this check requires using Guava which we don't want to use
-				"StringSplitter",
-
-				// Produces a lot of findings that we consider to be false positives, for example for package-private
-				// classes and methods
-				"MissingSummary",
+				"InlineMeSuggester",
+				"MissingSummary", // Produces a lot of findings that we consider to be false positives, for example for package-private classes and methods.
+				"StringSplitter", // We don't want to use Guava.
+				"UnnecessaryLambda", // The findings of this check are subjective because a named constant can be more readable in many cases.
+				// error-prone.picnic (https://error-prone.picnic.tech)
+				"ConstantNaming",
+				"FormatStringConcatenation",
+				"IdentityConversion",
+				"LexicographicalAnnotationAttributeListing",
+				"LexicographicalAnnotationListing",
+				"NestedOptionals",
+				"NonStaticImport",
+				"OptionalOrElseGet",
+				"PrimitiveComparison",
+				"StaticImport",
+				"TimeZoneUsage",
```

Comment:
These are failing and cannot be fixed through patching, which is somewhat of a mystery.